internal/edgeapi/yggdrasil/yggdrasil.go

```diff
@@ -159,17 +116,13 @@ func (h *Handler) GetControlMessageForDevice(ctx context.Context, params yggdras
 		return operations.NewGetControlMessageForDeviceInternalServerError()
 	}
 
-	if edgeDevice.DeletionTimestamp != nil && !utils.HasFinalizer(&edgeDevice.ObjectMeta, YggdrasilWorkloadFinalizer) {
-		if utils.HasFinalizer(&edgeDevice.ObjectMeta, YggdrasilConnectionFinalizer) {
-			err = h.deviceRepository.RemoveFinalizer(ctx, edgeDevice, YggdrasilConnectionFinalizer)
-			if err != nil {
-				return operations.NewGetControlMessageForDeviceInternalServerError()
-			}
-			h.metrics.IncEdgeDeviceUnregistration()
-		}
+	switch regStatus {
```

Comment:
is there anticipation for it to grow or can it be replaced by `if`?

Reply:
No, but it felt more go-ish than `if`.